src/utils/time-code.js

```diff
@@ -4,15 +4,38 @@
 
 // @flow
 
+const MAX_TIMINGS_PER_LABEL = 3;
+const _timingsPerLabel = {};
+
+/**
+ * We care about timing information. This function helps log and collect information
+ * about how fast our functions are.
+ */
 export function timeCode<T>(label: string, codeAsACallback: () => T): T {
-  if (
-    typeof performance !== 'undefined' &&
-    process.env.NODE_ENV === 'development'
-  ) {
+  if (typeof performance !== 'undefined') {
     const start = performance.now();
     const result = codeAsACallback();
     const elapsed = Math.round(performance.now() - start);
-    console.log(`${label} took ${elapsed}ms to execute.`);
+
+    // Only log timing information in development mode.
+    if (process.env.NODE_ENV === 'development') {
+      console.log(`${label} took ${elapsed}ms to execute.`);
+    }
+
+    // Some portion of users will have timing information sent. Limit this further to
+    // only send a few labels per user.
+    const ga = self.ga;
+    if (ga && !(_timingsPerLabel[label] > MAX_TIMINGS_PER_LABEL)) {
```

Comment:
I think you want `_timingsPerLabel[label] < MAX_TIMINGS_PER_LABEL` (I mean, remove the `!` of course, and use `<` instead of the current `<=` because otherwise it happens once too many)

Reply:
Thanks, not sure why I had it that way.